src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,186 @@
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT  authenticates to the source GHES API. GH_PAT authenticates with the target api.";
+
+            var ghesApiUrl = new Option<string>("--ghes-api-url")
+            {
+                IsRequired = false,
+                Description = "The api endpoint for the hostname of your GHES instance. For example: https://api.myghes.com"
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var sourceRepo = new Option<string>("--source-repo")
+            {
+                IsRequired = true
+            };
+            var githubTargetOrg = new Option<string>("--github-target-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable."
+            };
+            var targetRepo = new Option<string>("--target-repo")
+            {
+                IsRequired = false,
+                Description = "Defaults to the name of source-repo"
+            };
+            var azureStorageConnectionString = new Option<string>("--azure-storage-connection-string")
+            {
+                IsRequired = true,
+                Description = "The connection string for the Azure storage account. For example: DefaultEndpointsProtocol=https;AccountName=myaccount;AccountKey=mykey;EndpointSuffix=core.windows.net"
+            };
+            var noSslVerify = new Option("--no-ssl-verify")
+            {
+                IsRequired = false
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(ghesApiUrl);
+            AddOption(githubSourceOrg);
+            AddOption(sourceRepo);
+            AddOption(githubTargetOrg);
+            AddOption(targetRepo);
+            AddOption(azureStorageConnectionString);
+            AddOption(noSslVerify);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, string, string, string, string, bool, bool>(Invoke);
+        }
+
+        public async Task Invoke(string ghesApiUrl, string githubSourceOrg, string sourceRepo, string githubTargetOrg, string targetRepo, string azureStorageConnectionString, bool noSslVerify = false, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            // Log all the parameters, except for the connection string, which is a secret
+            _log.LogInformation("Starting Migration Archives...");
+            _log.LogInformation($"GHES SOURCE ORG: {githubSourceOrg}");
+            _log.LogInformation($"GHES SOURCE REPO: {sourceRepo}");
+            _log.LogInformation($"GITHUB TARGET ORG: {githubTargetOrg}");
+            _log.LogInformation($"GHES API URL: {ghesApiUrl}");
```

Comment:
we should wrap this in an if and only log/display it if provided. In general thats what we do for most (all?) optional args on other commands.

Reply:
These are all required parameters or non-null